specs/AppLifecycle/StateNotifications/AppLifecycle StateNotifications.md

```diff
@@ -0,0 +1,389 @@
+| GUID_SESSION_USER_PRESENCE        | The user status associated with the application's session has changed.                                                                                                                                                 | `UserPresenceStatus`                                                   |
+| GUID_SYSTEM_AWAYMODE              | The system is entering or exiting away mode.                                                                                                                                                                           | `SystemAwayModeStatus` |
+
+The existing PowerManager class exposes only static properties and events: all members are valid to
+be brought over to the Reunion version. Many existing enumerated values are brought over as-is as
+well.
+
+```idl
+namespace Microsoft.Windows.System.Power
+{
+    // Clone to Windows.System.Power.BatteryStatus
+    enum BatteryStatus {
+        NotPresent,
+        Discharging,
+        Idle,
+        Charging
+    }
+
+    // Clone of Windows.System.Power.EnergySaverStatus
+    enum EnergySaverStatus {
+        Disabled,
+        Off,
+        On
+    }
+
+    // Clone of Windows.System.Power.PowerSupplyStatus
+    enum PowerSupplyStatus {
+        NotPresent,
+        Inadequate,
+        Adequate
+    }
+
+    // Status of the device's power source
+    enum PowerSourceKind {
+        AC,
+        DC,
+        ShortTerm
+    }
+
+    // State of the primary display
+    enum DisplayState {
+        Off,
+        On,
+        Dimmed
+    }
+
+    // Indicates whether the system is busy or idle.
+    enum SystemBusyState {
+        Idle,
+        Busy,
+    }
+
+    // Active power management personalities supported by the system
+    enum PowerSchemePersonality {
+        HighPerformance,
+        PowerSaver,
+        Balanced
+    }
+
+    // User presence indicator - whether the user is present or absent
+    // (active or idle)
+    enum UserPresenceStatus {
+        Present,
+        Absent
+    }
+
+    // Indicates whether the system is entering or exiting "away" mode
+    enum SystemAwayModeStatus {
+        Entering,
+        Exiting
+    }
+
+    // Largely a clone of the Windows.System.Power.PowerManager type
+    static runtimeclass PowerManager
+    {
+        // gets the device's battery status
+        static BatteryStatus BatteryStatus { get; };
+
+        // gets the device's energy-saver status
+        static EnergySaverStatus EnergySaverStatus { get; };
+
+        // gets the device's power supply status
+        static PowerSupplyStatus PowerSupplyStatus { get; };
+
+        // gets the total percentage of charge remaining from all batteries connected to
+        // the device
+        static Int32 RemainingChargePercent { get; };
```

Comment:
What if there is no battery?

Reply:
DISCUSSION: See above; need a resolution on behavior here.

Reply:
Clarified: if Battery::NotPresent, then RemainingChargePercent==100.